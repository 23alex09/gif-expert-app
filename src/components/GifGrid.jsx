import { useState, useEffect } from 'react'
import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { getGifs } from '../helpers/getGifs'
import PropTypes from 'prop-types'


//useEffect efecto secundarios -> cosas que queremos ejecutar en consecuencia de algun cambio
export const GifGrid = ( { category } ) => {

    const { images, isLoading } = useFetchGifs( category )

    return (
        <>
            <h3>{ category }</h3>
            {
                isLoading && ( <h2>Cargando...</h2> )
            }

            <div className="card-grid"> {/*className es el equivalente a class*/ }
                {
                    images.map( ( image ) => (
                        <GifItem key={ image.id } { ...image } />
                        //Esparcimos las propiedades de la image para no tener que andar poniendola una a una
                    ) )
                }
            </div>
        </>
    )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}