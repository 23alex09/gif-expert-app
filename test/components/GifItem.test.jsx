import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe( 'Test on GifItem component', () => {

    const title = 'Titulo de prueba';
    const url = 'https://url-de-prueba.es/gif.jpg';

    test( 'should match with the snapshot', () => {
        const { container } = render( <GifItem title={ title } url={ url } /> );
        expect( container ).toMatchSnapshot();
    } )

    test( 'should show the image with the title and the url', () => {
        render( <GifItem title={ title } url={ url } /> );
        // screen.debug();
        const { src, alt } = screen.getByRole( 'img' )
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    } )

    test( 'should show the title as text', () => {
        render( <GifItem title={ title } url={ url } /> );
        expect( screen.getByText( title ) ).toBeTruthy();
    } )
} )