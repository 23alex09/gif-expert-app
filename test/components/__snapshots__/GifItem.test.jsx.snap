// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GifItem component should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <p>
      Titulo de prueba
    </p>
    <img
      alt="Titulo de prueba"
      src="https://url-de-prueba.es/gif.jpg"
    />
  </div>
</div>
`;
